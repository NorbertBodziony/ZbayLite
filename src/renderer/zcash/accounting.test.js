import BigNumber from 'bignumber.js'

import accountingFactory from './accounting'

describe('accounting', () => {
  const zcashClient = {
    request: {
      'z_getbalance': jest.fn(async (address) => 2.432)
    }
  }

  const accounting = accountingFactory(zcashClient)

  it('fetches address balance', async () => {
    const address = 'testaddress'
    const balance = await accounting.balance(address)
    expect(balance).toBeInstanceOf(BigNumber)
    expect(balance).toMatchSnapshot()
    expect(zcashClient.request.z_getbalance).toHaveBeenCalledWith(address)
  })
})
