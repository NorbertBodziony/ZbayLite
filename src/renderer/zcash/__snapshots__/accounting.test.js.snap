// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accounting fetches address balance 1`] = `"2.432"`;
